packages/turf-isolines/index.ts

```diff
@@ -151,11 +157,35 @@ function rescaleIsolines(
     point[1] = point[1] * scaleY + y0;
   };
 
-  // resize and shift each point/line of the createdIsoLines
+  // Ok. We need to *clone* the returned isolines due to the behaviour of
+  // the marchingsquare library behaviour. Long story short, if we don't
+  // do this, the last point in some isoLines has the coordEach function
+  // run twice, leading to one point in the isoline rendering WAY out of
+  // position.
+  const clonedIsoLines: Feature<MultiLineString>[] = [];
   createdIsoLines.forEach((isoline) => {
+    clonedIsoLines.push(clone(isoline));
+  });
+
+  // resize and shift each point/line of the createdIsoLines
+  clonedIsoLines.forEach((isoline) => {
     coordEach(isoline, resize);
```

Comment:
on a second thought, maybe the resize function should just not mutate the values, but return a new array:
```js
const resize = (point: number[]) => [
  point[0] * scaleX + x0,
  point[1] * scaleY + y0,
];
```

Reply:
I think the resize function has to mutate in place - coordEach discards the return value of the callback.

It's more a quirk of the marchingsquares library which seems to re-use the first coordinate array as the closing point as well. So the callback passed to coordEach is being unexpectedly run twice on a single set of numbers i.e.

What we get back from marchingsquares:

[0] -> [1, 1]
[1] -> [2, 2]
[2] -> [3, 3]
[3] -> [1, 1] // <-- not another array that also contains [1, 1], the **exact same** array referred to by [0]

What we end up with after coordEach (if callback appends an "x" for the sake of visualisation):

[0] -> [1xx, 1xx]
[1] -> [2x, 2x]
[2] -> [3x, 3x]
[3] -> [1xx, 1xx] // in our case making one point way off target

I'll tweak the workaround to maybe only clobber the last coord in each linestring with a cloned value, then run the resize function over that.

Reply:
Looks like this might have been a red herring. Pulled out the cloning operation to rework it and the tests still passed. Suspect adding linearRing: false to marchingsquares.isoContours() may have fixed it already. CI tests have passed so will merge. Thanks for questioning it @stebogit 👍